versioning.cmake

```diff
@@ -0,0 +1,106 @@
+# -----------------------------------------------------------------------------
+# HTM Community Edition of NuPIC
+# Copyright (C) 2019, Numenta, Inc.
+#
+# Author: David Keeney, 2019
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+# -----------------------------------------------------------------------------
+
+
+# figures out the version from git and returns Version, Major, Minor, Patch
+#
```

Comment:
please extend the doc a bit, 
to which file it writes, that the file is modified automatically?, that it sets version from tag (note, the format is `v2.0.0-aa` currently, can also be just `v2.0.0`) 
That it's used in setup.py to set version, ...

Reply:
> please extend the doc a bit,

Sure.   Probably should refactor it a little too.